posthog/api/routing.py

```diff
@@ -81,13 +81,17 @@ def filter_queryset_by_parents_lookups(self, queryset):
     def get_parents_query_dict(self) -> Dict[str, Any]:
         if getattr(self, "_parents_query_dict", None) is not None:
             return cast(Dict[str, Any], self._parents_query_dict)
+
+        # used to override the last visited project if there's a token in the request
+        team_from_request = self._get_team_from_request()
+
         if self.legacy_team_compatibility:
             if not self.request.user.is_authenticated:
                 raise AuthenticationFailed()
             project = self.request.user.team
             if project is None:
                 raise ValidationError("This endpoint requires a project.")
-            return {"team_id": project.id}
+            return {"team_id": team_from_request.id if team_from_request else project.id}
```

Comment:
If we already have a `team_from_request`, then we shouldn't need to access `self.request.user.team` at all?

Reply:
makes sense